api/api.go

```diff
@@ -130,10 +131,108 @@ func (a *API) ExecuteTask(serviceID, taskKey string, inputData map[string]interf
 	if err = a.execDB.Save(exec); err != nil {
 		return "", err
 	}
-	go a.ps.Pub(exec, s.TaskSubscriptionChannel())
+	go a.ps.Pub(exec, s.ExecutionSubTopic())
 	return exec.ID, nil
 }
 
+// ListenEvent listens events matches with eventFilter on serviceID.
+func (a *API) ListenEvent(service string, f *EventFilter) (*EventListener, error) {
+	s, err := a.db.Get(service)
+	if err != nil {
+		return nil, err
+	}
+
+	if f.HasKey() {
+		if _, err := s.GetEvent(f.Key); err != nil {
+			return nil, err
+		}
+	}
+
+	l := NewEventListener(a.ps, s.EventSubTopic(), f)
+	go l.Listen()
+	return l, nil
+}
+
+// ListenExecution listens executions on service.
+func (a *API) ListenExecution(service string, f *ExecutionFilter) (*ExecutionListener, error) {
+	s, err := a.db.Get(service)
+	if err != nil {
+		return nil, err
+	}
+
+	if f != nil {
+		if f.TaskKey == "" && f.OutputKey != "" {
+			return nil, fmt.Errorf("execution filter: output key given without task key")
+		}
+		if f.HasTaskKey() {
+			task, err := s.GetTask(f.TaskKey)
+			if err != nil {
+				return nil, err
+			}
+			if _, err := task.GetOutput(f.OutputKey); f.OutputKey != "" && err != nil {
```

Comment:
The output key should also accept `*`.
See previous implementation:
https://github.com/mesg-foundation/core/blob/d7500d9251bb5e562e9ac98d76607f558a5f100c/api/listen_result.go#L128-L138

I think you should add a function `HasTaskOutputKey` similar to `HasTaskKey`.

To reproduce:
```
mesg-core service dev https://github.com/mesg-foundation/service-ethereum-erc20 -t name -o=*
⨯ Listening results error: rpc error: code = Unknown desc = Output "*" of task "name" not found in service "Ethereum ERC20 token"
```